testsuite/databases/pgsql/discover.py

```diff
@@ -67,7 +67,7 @@ class PgShardedDatabase:
 
 
 def find_schemas(
-        service_name: str, schema_dirs: List[pathlib.Path],
+        service_name: Optional[str], schema_dirs: List[pathlib.Path],
```

Comment:
why do you want it to be optional?

Reply:
There are 2 reasons:

1. The documentation says that it is optional - service_name: service name used as prefix for database name if not empty, e.g. "servicename_dbname". If it is empty string, then the result will be "_dbname", and None gives expected output "dbname"
2. I have a use case when I need to pass None. I can send you a link to PR in arc.

Reply:
okk, so make it optional for _find_databases_schemas as well